### PR TITLE
Additional error reporting around safe links

### PR DESCRIPTION
**Pull request by sp1ff**
This commit adds an additional log message (at level ERR) explaining why we refuse to follow a symlink in certain conditions. I was confused when it happened to me, and this error message would have been a big help.

It also adds GNU global tags, as well as systemd files, to the relevant .gitingore files.

The PR contains my commit as well as your merge of my last PR. Let me know if that's a problem.

### CODE REVIEW

**Comment by sp1ff** [comment]
I confess, I have _no_ idea why I was able to get it down to one commit, this time...

File: misc/systemd/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+cf-*.service
+cfengine3.service
```
Comment (vpodzime) [inline misc/systemd/.gitignore]: These are now in the repo-wide `.gitignore` file.

File: .gitignore [hunk 2]
```diff
@@ -92,6 +92,9 @@ coverage-html/
 TAGS
 tags
 cscope.*
+GPATH
+GRTAGS
+GTAGS
```
Comment (vpodzime) [inline .gitignore]: This should go into a separate commit, it has nothing to do with the change below.

**Comment by vpodzime** [comment]
@sp1ff could you please address the above comments so that this can get merged? Thanks!

**Comment by vpodzime** [comment]
> @vpodzime Sorry-- day job. Should now be ready to merge.

No problem. Thanks!
